bdk_chain/src/tx_graph.rs

```diff
@@ -167,38 +170,87 @@ impl TxGraph {
         self.txs.iter().filter_map(|(txid, tx)| match tx {
             TxNode::Whole(_) => None,
             TxNode::Partial(partial) => Some((*txid, partial)),
         })
     }
 
-    /// Return an iterator of conflicting txids, where the first field of the tuple is the vin of
-    /// the original tx in which the txid conflicts.
-    pub fn conflicting_txids<'g>(
+    /// Creates an iterator that both filters and maps descendants of the given `txid`.
+    ///
+    /// When the supplied closure returns `None`, spend paths from the given descendant are not
+    /// explored, and the descendant will not be yielded by the iterator.
+    pub fn filter_map_descendants<'g, T>(
+        &'g self,
+        txid: Txid,
+        filter_map: impl FnMut(OutPointSpends, Txid) -> Option<T> + 'g,
+    ) -> impl Iterator<Item = T> + '_ {
+        TxDescendants::new(self, txid, filter_map)
+    }
+
+    /// Creates an iterator that both filters and maps conflicting transactions (this includes
+    /// descendants of adjacently-conflicting transactions).
+    ///
+    /// When the supplied closure returns `None`, spend paths from the given descendant are not
+    /// explored, and the descendant will not be yielded by the iterator.
+    pub fn filter_map_conflicts<'g, T: 'g>(
+        &'g self,
+        tx: &'g Transaction,
+        filter_map: impl Fn(OutPointSpends, Txid) -> Option<T> + 'g,
+    ) -> impl Iterator<Item = T> + '_ {
+        let spends = tx.input.iter().filter_map(|txin| {
+            self.spends
+                .get(&txin.previous_output)
+                .map(|spends| (txin.previous_output, spends))
+        });
+
+        TxDescendants::from_spends(self, spends, [tx.txid()].into(), filter_map)
+    }
+
+    /// Given a `txin` and the `txid` of the transaction in which it resides, return an iterator of
+    /// txids which directly conflict with the given `txin`.
+    ///
+    /// **WARNING:** It is up to the caller to ensure the given `txin` actually resides in `txid`.
+    /// **WARNING:** This only returns directly conflicting txids and does not include descendants
+    /// of those txids.
+    pub fn adjacent_conflicts_of_txin<'g>(
+        &'g self,
+        txid: Txid,
+        txin: &'g TxIn,
+    ) -> impl Iterator<Item = Txid> + '_ {
+        self.spends
+            .get(&txin.previous_output)
+            .into_iter()
+            .flatten()
+            .filter(move |&&conflicting_txid| conflicting_txid != txid)
+            .cloned()
+    }
+
+    /// Given a transaction, return an iterator of txids which directly conflict with the given
+    /// transaction's inputs (spends). The conflicting txids are returned with the given
+    /// transaction's vin (in which it conflicts).
+    ///
+    /// **WARNING:** This only returns directly conflicting txids and does not include descendants
+    /// of those txids.
+    pub fn adjacent_conflicts_of_tx<'g>(
```

Comment:
```suggestion
    pub fn direct_conflicts_of_tx<'g>(
```
You use the term direct in the doc comment to describe this. I think you should call it that.